x64: implement add/sub with wrapping and xor op

kubkon opened the pull request.

andrewrk commented on src/arch/x86_64/CodeGen.zig:
```diff
@@ -1318,7 +1318,7 @@ fn airAddWrap(self: *Self, inst: Air.Inst.Index) !void {
         .dead
     else
-        return self.fail("TODO implement addwrap for {}", .{self.target.cpu.arch});
+        try self.genBinMathOp(inst, bin_op.lhs, bin_op.rhs);
```
this function now is identical to `airAdd`, correct? I suggest to delete one of them and have the big switch map both `add` and `add_wrap` to the same function.

kubkon replied:
Done!

kubkon commented:
While addressing the feedback, I noticed that `stage1/obj/constant_inside_comptime_function_has_compile_error.zig` is failing with a wrong compile error - any ideas if this is a potential regression or whatnot?

andrewrk commented:
Hmm yes I see that running the example locally produces:

```
[nix-shell:~/dev/zig/build-release]$ ./zig build-obj ../test/compile_errors/stage1/obj/constant_inside_comptime_function_has_compile_error.zig 
../test/compile_errors/stage1/obj/constant_inside_comptime_function_has_compile_error.zig:3:28: error: unused function parameter
pub fn MemoryPool(comptime T: type) type {
                           ^
../test/compile_errors/stage1/obj/constant_inside_comptime_function_has_compile_error.zig:12:9: error: unused local variable
    var allocator: ContextAllocator = undefined;
        ^
```

However this does not match the expected compile errors. ~And yet running the compile error tests succeeds. So I think something is wrong with the test harness.~

I get this when running the test harness:

```
$ ./zig build test-stage2 -fqemu -Denable-llvm -fwasmtime 
Test [1/104] test ""... FAIL (InvalidFile)
/home/andy/dev/zig/src/test.zig:713:17: 0x2a8bb4c in TestContext.addErrorCasesFromDir (test)
                return error.InvalidFile; // Manifests are currently mandatory
                ^
/home/andy/dev/zig/test/compile_errors.zig:30:13: 0x2a87246 in .test_cases.compile_errors.addCases (test)
            try ctx.addErrorCasesFromDir("stage1", obj_dir, .stage1, .Obj, false, one_test_case_per_file);
            ^
/home/andy/dev/zig/test/cases.zig:10:5: 0x2a833f7 in .test_cases.addCases (test)
    try @import("compile_errors.zig").addCases(ctx);
    ^
/home/andy/dev/zig/src/test.zig:34:5: 0x2a5e346 in test "" (test)
    try @import("test_cases").addCases(&ctx);
    ^
103 passed; 0 skipped; 1 failed.
error: the following test command failed with exit code 1:
/home/andy/dev/zig/zig-cache/o/e846cdc598f60a6e7f170291c16a609a/test /home/andy/dev/zig/build-release/zig
```

Not sure when this started failing. Next step here is to improve `addErrorCasesFromDir` to print the file that is missing the manifest.

andrewrk commented:
Ah much better:

```
$ ./zig build test-stage2 -fqemu -Denable-llvm -fwasmtime 
Test [1/104] test ""... thread 1310090 panic: test harness failed to process file '.constant_inside_comptime_function_has_compile_error.zig.swp': InvalidFile
```

Now I'll just ignore unknown file extensions...

And finally, the test suite agrees with you!

```
$ ./zig build test-stage2 -fqemu -Denable-llvm -fwasmtime 
Test [1/104] test ""... compiler [1/852] stage2: slice of null C pointer [1/1] update [2/3] AST Lowerincompiler [129/852] 
=========== Expected compile error: ============
tmp.zig:4:5: error: unreachable code
================= Full output: =================
./zig-cache/tmp/81wQlhnlS-Ymov0H/tmp.zig:3:28: error: unused function parameter
./zig-cache/tmp/81wQlhnlS-Ymov0H/tmp.zig:12:9: error: unused local variable

================================================
test 'stage1: constant inside comptime function has compile error' failed: TestFailed
```

andrewrk commented:
Alright you should be unblocked if you rebase against latest master. Sorry for the trouble - root cause was a regression I caused with cf4aad4858ac61b4814d8f021c8eae22ee7f63e6.
